src/Core/Layer/LayerUpdateState.js

```diff
@@ -57,10 +58,16 @@ LayerUpdateState.prototype.success = function success() {
 };
 
 LayerUpdateState.prototype.noMoreUpdatePossible = function noMoreUpdatePossible() {
+    this.failureParams = undefined;
     this.state = UPDATE_STATE.FINISHED;
 };
 
-LayerUpdateState.prototype.failure = function failure(timestamp, definitive) {
+LayerUpdateState.prototype.lastFailureParams = function _lastFailureParams() {
+    return this.failureParams[this.failureParams.length - 1];
```

Comment:
Since you're now only ever using the last failure parameters, there's no need to keep an array anymore.

And maybe you could reset (`this.failureParams = undefined;) it on success as well?